backend/app/schemas/time_block.py

```diff
@@ -0,0 +1,24 @@
+
+from pydantic import BaseModel
+from datetime import datetime
+from uuid import UUID
+
+class TimeBlockBase(BaseModel):
+    start_time: datetime
+    end_time: datetime
+
+class TimeBlockId(BaseModel):
+    id: UUID
+
+class TimeBlockFull(TimeBlockBase, TimeBlockId):
+    '''
+    Combines TimeBlockBase and TimeBlockId.
+    Represents a full time block with an ID and time range.
+    '''
+    pass
+
+class TimeBlockInDB(BaseModel):
```

Comment:
Call this `TimeBlockEntity` instead, `InDB` isn't standard for DB representations.